datasource.go

```diff
@@ -0,0 +1,204 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/bitly/go-simplejson"
+	"github.com/grafana/grafana-plugin-model/go/datasource"
+	plugin "github.com/hashicorp/go-plugin"
+	"golang.org/x/net/context"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+var httpClient = &http.Client{}
+
+type ClickhouseDatasource struct {
+	plugin.NetRPCUnsupportedPlugin
+}
+
+func (t *ClickhouseDatasource) Query(ctx context.Context, req *datasource.DatasourceRequest) (r *datasource.DatasourceResponse, err error) {
```

Comment:
Named return is used to return the error on panic. I am not using the name for the response, but I have included it because go requires them all to be named or unnamed

Reply:
Technically, the hashicorp plugin wraps the module and outputs panics to the error log. However, the caller must supply a log function for it to be logged and AFAIK grafana does not do it.